mesonpy/_tags.py

```diff
@@ -1,130 +1,124 @@
+    name = sys.implementation.name
+    name = INTERPRETERS.get(name, name)
+    version = sys.version_info
+    return f'{name}{version[0]}{version[1]}'
+
+
+def _get_config_var(name: str, default: Union[str, int, None] = None) -> Union[str, int, None]:
+    value = sysconfig.get_config_var(name)
+    if value is None:
+        return default
+    return value
+
+
+def _get_cpython_abi() -> str:
+    version = sys.version_info
+    debug = pymalloc = ''
+    if _get_config_var('Py_DEBUG', hasattr(sys, 'gettotalrefcount')):
+        debug = 'd'
+    if sys.version_info < (3, 8) and _get_config_var('WITH_PYMALLOC', True):
+        pymalloc = 'm'
+    return f'cp{version[0]}{version[1]}{debug}{pymalloc}'
+
+
+def get_abi_tag() -> str:
+    abi = sysconfig.get_config_var('SOABI')
+    if not abi:
+        # CPython on Windows does not have a SOABI sysconfig variable.
+        assert sys.implementation.name == 'cpython'
+        return _get_cpython_abi()
+    if abi.startswith('cpython'):
+        return 'cp' + abi.split('-')[1]
+    if abi.startswith('pypy'):
+        return '_'.join(abi.split('-')[:2])
+    return abi.replace('.', '_').replace('-', '_')
+
+
+def _get_macosx_platform_tag() -> str:
+    ver, x, arch = platform.mac_ver()
+
+    # Override the macOS version if one is provided via the
+    # MACOS_DEPLOYMENT_TARGET environment variable.
+    try:
+        version = tuple(map(int, os.environ.get('MACOS_DEPLOYMENT_TARGET', '').split('.')))[:2]
+    except ValueError:
+        version = tuple(map(int, ver.split('.')))[:2]
+
+    # Python built with older macOS SDK on macOS 11, reports an
+    # unexising macOS 10.16 version instead of the real version.
+    #
+    # The packaging module introduced a workaround
+    # https://github.com/pypa/packaging/commit/67c4a2820c549070bbfc4bfbf5e2a250075048da
+    #
+    # This results in packaging versions up to 21.3 generating
+    # platform tags like "macosx_10_16_x86_64" and later versions
+    # generating "macosx_11_0_x86_64".  Using latter would be more
+    # correct but prevents the resulting wheel from being installed on
+    # systems using packaging 21.3 or earlier (pip 22.3 or earlier).
+    #
+    # Fortunately packaging versions carrying the workaround still
+    # accepts "macosx_11_0_x86_64" as a compatible platform tag.  We
+    # can therefore ignore the issue and generate the slightly
+    # incorrect tag.
+
+    major, minor = version
+
+    if major >= 11:
+        # For macOS reelases up to 10.15, the major version number is
+        # actually part of the OS name and the minor version is the
+        # actual OS release.  Starting with macOS 11, the major
+        # version number is the OS release and the minor version is
+        # the patch level.  Reset the patch level to zero.
+        minor = 0
+
+    if sys.maxsize <= 2**32:
+        # 32-bit Python running on a 64-bit kernel.
+        if arch.startswith('ppc'):
+            arch = 'ppc'
+        arch = 'i386'
```

Comment:
It looks like `arch = 'i386'` will overwrite `arch = 'ppc'`; should that be before the conditional, or guarded by `else:`?

Reply:
Good catch! I guess I refactored this code around and I had this in a function with a `return` before.